### PR TITLE
Add TLS1.2 workaround for code coverage script

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

GitHub needs TLS1.2. Invoke-WebRequest defaults do not support TLS1.2 and hence code coverage runs are blocked as they cannot get artifacts from GitHub. 

This PR adds in the workaround to change the SecurityProtocol to add TLS1.2

Also added `-ErrorAction SilentlyContinue` so it does not throw a non-terminating error when `pwsh.exe` is not found.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [X] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by JamesWTruher — APPROVED** [review]

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 1]
```diff
@@ -145,8 +145,16 @@ $jsonFile = "$outputBaseFolder\CC.json"
 
 try
 {
+    ## Github needs TLS1.2 whereas the defaults for Invoke-WebRequest do not have TLS1.2
+    $prevSecProtocol = [System.Net.ServicePointManager]::SecurityProtocol
+
+    [System.Net.ServicePointManager]::SecurityProtocol =
+        [System.Net.ServicePointManager]::SecurityProtocol -bor
```
Comment (TravisEz13) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: don't or it with the existing setting.  The SDL say 1.0 and newer at most.

**Review by TravisEz13 — APPROVED** [review]

**Comment by markekraus** [comment]
Hmm... I think we should look at add this more globally. I was under the impression that Start-PSBootstrap would be run in all of the CIs before any commands and we wouldn't need to make this change in every location. One the recent PRs should have addresses this. 

If we have a common entry point for all activities in all environments, we should make this setting change there. perhaps setting it during import of build.psm1? 

Separate issue: consider a patch to Windows PowerShell 5.1 to enable TLS 1.1 and TLS 1.2 by default.

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 (hunk 1 above)
Comment (adityapatwardhan) [reply test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: The default is TLS 1.0.

**Comment by adityapatwardhan** [comment]
@markekraus This particular script does not run in CI, but a different test machine as the execution time for code coverage runs exceed CI run time limits. 

Also, this script does not use `Start-PSBootstrap` as it does not build PowerShell but uses the packages generated by AppVeyor nightly build.

**Pull request merged**